### PR TITLE
Zijie lint fix for reports component Phase 1

### PR DESCRIPTION
**Pull request by iTvX**
# Description
Lint fix for Reports component, this is phase 1.

## Files changed

- BadgeSummaryPreview.jsx
- Badgesummaryviz.jsx  
- InfringementsViz.jsx
- PeopleTable.js

# How to Test

To test this pull request:

- Check out the current branch
- Run `npm install` and `npm run start:local` to run this PR locally  
- Log in as an admin user
- Go to Reports > People > select a user > click "Show Badge",  Check functionality is the same as before. (Please record a video)
- Check other functionality of the report is the same as before, and see if any bugs have appeared. (Please record a video)

Optional:
- Set up ESLint in VSCode/ IntelliJ IDEA, I personally use IntelliJ IDEA
- `cd src/components/Reports`
  - Run `npx eslint `for the files: BadgeSummaryPreview.jsx, BadgesummaryViz.jsx, InfringementsViz.jsx, PeopleTable.js
  - Check if you find any errors
I have complete confidence that **the code will pass the ESLint tests.** Therefore, I kindly request that you **focus on testing the frontend**. Specifically, please ensure that the **report functionality remains consistent** with how it operated previously. Thank you for your time and effort!

Screenshot:
CleanShot 2023-09-27 at 02 47 27@2x

## Note:
Include the information the reviewers need to know.

### CODE REVIEW

**Comment by luisarevalo21** [comment]
Hello @iTvX,

I tested your branch and works as intended great work! 
Upon running the npx eslint on each file no errors occur. 
I only get a warning as you showed in your example. 
However, I'm a bit confused on what to test when the app is running. Checking the user's badges works as normal but your comment says "Check other reports' function" do you mean just run the app as normal? Or is there anything else specific I should test. 



Screenshot 2023-09-27 at 7 34 56 PM

**Comment by DavidC0126** [comment]
Hi @iTvX I tested all 4 files you mentioned in src/component/Reports folder, and there is no any error pop up. Good Job! The functions work in the web pages too.
no_error
show_badges

**Comment by iTvX** [comment]
Hi @luisarevalo21 , thank you for testing. 

Yes, you are correct. Just run the app as normal, if you didn't see any error or bugs, could you please approve this PR?

**Comment by Jiyuanxia** [comment]
Hi @iTvX , I have reviewed your pull request and tested it. Functionality is the same as before but after I selected a user, some error messages appear. And I only get a warning as you showed.
Screenshot 2023-09-28 183015
Screenshot 2023-09-28 183028
Screenshot 2023-09-28 183247

**Review by tsunami776 — APPROVED** [review]
Hi @iTvX I followed your steps in the PR, and I ran `npx lint` for the files in `src/component/Reports`. No error was prompted. I tested with admin account for 'Show Badges' functionalities, and it works as intended. Great work! I tested dev branch too, and It seems the error message mentioned above also appears in dev branch. 




dev branch:

**Comment by iTvX** [comment]
@tsunami776  Thank you so much for testing the dev branch and approving this PR.

**Comment by iTvX** [comment]
@Jiyuanxia Thank you for testing. I checked the dev branch and found the same error. So, it should not be an issue caused by this pull request, but thank you very much!

**Review by Sssky-18 — DISMISSED** [review]
Hi @iTvX, I tested this PR, and it worked as expected. Firstly, the pages showed normally, and the badges were also displayed the same as that before. Besides, the eslint fix worked as you stated in the example. So I approved this PR. 

PR#1334_2
PR#1334_3

**Review by Haoj1 — COMMENTED** [review]
@iTvX Hi I have tested your pr and have the same warnings as mentioned. But there is no errors when run npx eslint

**Review by zuhang2000 — COMMENTED** [review]
Hi I've tested your PR following your instructions. The "show badges" feature is working as expected. 
Screenshot 2023-09-30 at 11 15 48 PM
However, I do encounter this when running "npx eslint InfringementsViz.jsx". Please check the screenshot below to see if this can be fixed.
Screenshot 2023-09-30 at 11 18 43 PM

**Review by sxiong5 — COMMENTED** [review]
Hey @iTvX, I have tested the PR, and there are not eslint errors. However, there ara so many errors in the console of the browser which doesn't follow the React and HTML standard. Maybe consider fixing these will be better.

Screenshot 2023-10-05 at 16 39 26 (2)

**Review by Shiwani99 — APPROVED** [review]
Hi,
Great work. I tested your PR and it works as expected. I did both frontend check for the reports page and lint check in terminal. I checked every functionality in reports page and it worked well.

Screenshot 2023-10-06 at 6 30 08 PM


https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/78906820/fa7ca2c9-5b9d-49c2-8a9f-ec9d874d8585

**Review by Shubhu0 — COMMENTED** [review]
Hi, @iTvX Upon reviewing and testing your branch, the "Show Badges" is working as intended. 

image

Upon running npx eslint on the files, no error was prompted. Good Work!

image

**Comment by savio-henrique** [comment]
Hi @iTvX! I've reviewed and tested this branch and it worked as expected! And it showed no errors upon linting the files, only warnings.

terminal

**Review by savio-henrique — APPROVED** [review]
Hi @iTvX! I've reviewed and tested this branch and it worked as expected! And it showed no errors upon linting the files, only warnings.

terminal

**Review by Shubhu0 — DISMISSED** [review]
Hi @iTvX tested all 4 files and found no major errors!

image

File: src/components/Reports/BadgeSummaryPreview.jsx [hunk 1]
```diff
@@ -1,6 +1,7 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (EvianTan) [inline src/components/Reports/BadgeSummaryPreview.jsx:1]: I pulled your branch on my local and deleted this line 1 and ran `npx eslint <this_fine_name>`, found out it just complained about "React is defined but never used".

Please just delete your line 1 and delete the React import and apply such change to the repo.

File: src/components/Reports/BadgeSummaryViz.jsx [hunk 2]
```diff
@@ -1,3 +1,4 @@
+// eslint-disable-next-line no-unused-vars
```
Comment (EvianTan) [inline src/components/Reports/BadgeSummaryViz.jsx:1]: same as my comment on `src/components/Reports/BadgeSummaryPreview.jsx`

**Review by EvianTan — COMMENTED** [review]
Please address my two comments above

**Review by yi-lin-1234 — DISMISSED** [review]
Hello, I tested your PR and works as intended. Great work!
Screenshot 2023-10-22 at 4 01 39 PM
Screenshot 2023-10-22 at 4 05 04 PM

**Comment by iTvX** [comment]
> Please address my two comments above
Hi @EvianTan ,

Thank you, this issue has been fixed.

**Review by EvianTan — APPROVED** [review]

**Pull request merged**